### ISSUE TITLE
[a11y] Heading of the World Cup widget on HNT is missing a heading trait

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#34194]
## Steps to reproduce
1. Have VoiceOver running
1. Open a new tab and confirm the World Cup widget is present between the Shortcuts and News sections
1. Navigate to the visible heading of the widget, i.e. (currently) "Countdown to the World Cup" and observe the VO announcement
1. Use VO Rotor > Headings and attempt to get to the World Cup's heading via the Rotor actions

### Expected behavior
1. The on-screen heading is announced as "heading" by the VO, i.e. "Countdown to the World Cup, heading"
1. VO Rotor allows to get to this heading via its actions

### Actual behavior
1. The on-screen heading is announced as a static text by the VO without providing any role, i.e. "Countdown to the World Cup"
1. VO Rotor skips the WC heading while navigating between headings on the page
1. **Outcome: User may not be aware of this new widget existing on their HNT and won't be able to quickly access it with VoiceOver**. While it is a technically `access-S3` bug, I want to highlight that the heading structure is an important efficiency hook for assistive technology users. Per [the WebAIM Screen Reader User Survey #10 results](https://webaim.org/projects/screenreadersurvey10/#problematic), `Missing or improper headings` were included in Top-10 list of problematic items in the overall rating of difficulty and frustration for screen reader users. [80% of the responders are relying on some form of markup to find information on a lengthy page](https://webaim.org/projects/screenreadersurvey10/#finding), among them 71.6% of responders are likely to use headings first, thus this bug is a paper cut that would likely be noticeable by users.

### Device & build information
* Device: iPhone 15 Pro running iOS 26.5 
* Build version: Firefox 151.2 (72455) 
* First seen version: ? 
### Notes
Attachments: 

<img width="445" height="902" alt="HNT WC widget with VO focusing the implicit heading but the output captions not mentioning it is a heading" src="https://github.com/user-attachments/assets/17b43193-3e82-42e2-9b59-fb2ab37f579f" />

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-16028)
